data/migrations/20190321213238_createQuantitiesTable.js

```diff
@@ -0,0 +1,37 @@
+exports.up = function(knex, Promise) {
+  return knex.schema.createTable("quantities", tbl => {
+    tbl.increments();
+    tbl
+      .float("amount")
+      .unsigned()
+      .notNullable();
+    tbl
+      .integer("unit_id")
+      .notNullable()
+      .unsigned()
+      .references("id")
+      .inTable("units")
+      .onDelete("CASCADE")
+      .onUpdate("CASCADE");
+    tbl
+      .integer("ingredient_id")
+      .notNullable()
+      .unsigned()
+      .references("id")
+      .inTable("ingredients")
+      .onDelete("CASCADE")
+      .onUpdate("CASCADE");
+    tbl
+      .integer("recipe_id")
+      .notNullable()
+      .unsigned()
+      .references("id")
+      .inTable("recipes")
+      .onDelete("CASCADE")
+      .onUpdate("CASCADE");
```

Comment:
These foreign keys all look just about the same; why not make a helper function to break out that logic? It could look something like this:
```javascript
const foreignId = (tbl, tableName, single) => {
  tbl
      .integer(`${single}_id`)
      .notNullable()
      .unsigned()
      .references('id')
      .inTable(tableName)
      .onDelete("CASCADE")
      .onUpdate("CASCADE");
};

exports.up = // ...
  tbl.float('amount') // ...
  foreignId(tbl, 'units', 'unit');
  // other two foreign ids
```

Reply:
This suggestion also worked and helped make things DRYer.  Thanks for the feedback!